src/org/daisy/dotify/formatter/impl/volume/TocSequenceEventImpl.java

```diff
@@ -164,8 +167,53 @@ public BlockSequence getBlockSequence(FormatterContext context, DefaultContext v
 
 	private Predicate<String> refToVolume(Integer vol, CrossReferenceHandler crh) {
 		return refId -> {
-			Integer volNo = crh.getVolumeNumber(refId);
-			return vol == null ? volNo == null : vol.equals(volNo);
+			VolumeData volumeData = crh.getVolumeData(refId);
+			if (vol == null || volumeData == null) {
+				return vol == null && volumeData == null;
+			} else {
+				return vol.equals(volumeData.getVolumeNumber());
+			}
+		};
+	}
+	
+	/**
+	 * Determines whether a range is part of a volume
+	 * 
+	 * @param vol volume
+	 * @param crh cross-reference handler
+	 * @return 
+	 */
+	private Predicate<TocEntryOnResumedRange> rangeToVolume(Integer vol, CrossReferenceHandler crh) {
+		return range -> {
+			/* startVolumeData refers to the location where the range starts */
+			VolumeData startVolumeData = crh.getVolumeData(range.getStartRefId());
+			if (vol == null || startVolumeData == null) {
+				if (vol != null || startVolumeData != null) {
+					return false;
```

Comment:
This doesn't make much sense. There is no reason to render a `toc-entry-on-resumed` if the start of the range does not fit into the reserved volumes (`startVolumeData == null`), and to omit it if the range starts within a volume.

Actually rendering `toc-entry-on-resumed` when `vol == null` makes little sense altogether I think. We render the `toc-entry` because we know for sure that eventually they will be rendered (and because we wanted to mimic the old behavior as much as possible). But for `toc-entry-on-resumed` we really can not make a sensible prediction.

I would simply replace the `rangeToVolume(null, crh)` above with `x -> false`, and then the `rangeToVolume` function can also be simplified a lot.

Reply:
I did it in this way because of symmetry. If you think it doesn't make sense to render a `toc-entry-on-resumed` when `vol==null`, then also it shouldn't make sense to render a `toc-entry` when `vol==null`. In method `refToVolume` a `toc-entry` can be rendered when `vol==null`.

Reply:
Well it's clearly two different cases isn't it? Just think for a moment about what this code actually does.

Yes, rendering `toc-entry` when `vol == null` is indeed also questionable, we knew that. But it's less questionable than in the case of `rangeToVolume` and there is a clear purpose (as I explained in my second paragraph). For `rangeToVolume` I see absolutely no benefit in handling `vol == null`. It's just noise basically.

Reply:
I will make the change because it indeed simplifies the code. However we should realise that this results in a non-symmetric piece of code, which I don't like.

Reply:
As far as I'm concerned we can eventually also change `refToVolume` the same way to "restore symmetry". That means Daniel will have to update his regression test suite though.